### DISCUSSION TITLE
Add Lap Number to Car and Location data feed

### DISCUSSION THREAD

**Discussion by gmcfalls** [br-g/openf1#66] [Ideas]
it would be nice to filter the car and location data by lap number without having to use the date field

**Comment by PrestonHager** [comment]
I think the reason there's not a lap field associated with each car data entry is partially the size of the database. This would require a lot of database requests as the data is stored in two objects, one for car data, the other for lap data. Likewise, the API would need to do what you're probably currently doing and get the lap data start and next lap start then filter car data between those two dates, and finally add the lap number. When querying a large amount of data this is impractical and unnecessary since most users don't need lap number in the car data.

Instead, in your application, find out what lap number you need, then query the `/laps` endpoint for the given lap number of any driver and the next lap number. Then use the start dates as bounds for a query to the `car_data` endpoint. For Python this would look something like this:

```python
lap_data_8_9 = requests.get("https://api.openf1.org/v1/laps?session_key=9161&driver_number=2&lap_number>=8&lap_number<=9").json()
for i in range(len(lap_data_8_9)):
  set_attr(f"lap_{lap_data_8_9[i]['lap_number']}_start", lap_data_8_9[i]["date_start"])

# Then get the car data for that driver based on the dates
car_data = requests.get(f"https://api.openf1.org/v1/car_data?driver_number=2&session_key=9161&date>={lap_8_start}&date<={lap_9_start}").json()
```

**Reply by br-g** [reply]
Thanks a lot Preston!
